store/mockstore/mocktikv/mvcc_leveldb.go

```diff
@@ -931,6 +941,87 @@ func (mvcc *MVCCLevelDB) Cleanup(key []byte, startTS uint64) error {
 	return mvcc.db.Write(batch, nil)
 }
 
+// CheckTxnStatus checks the primary lock of a transaction to decide its status.
+// The return values are (ttl, commitTS, err):
+// If the transaction is active, this function returns the ttl of the lock;
+// If the transaction is committed, this function returns the commitTS;
+// If the transaction is rollbacked, this function returns (0, 0, nil)
+// Note that CheckTxnStatus may also push forward the `minCommitTS` of the
+// transaction, so it's not simply a read-only operation.
+func (mvcc *MVCCLevelDB) CheckTxnStatus(primaryKey []byte, startTS uint64, currentTS uint64) (uint64, uint64, error) {
+	mvcc.mu.Lock()
+	defer mvcc.mu.Unlock()
+
+	startKey := mvccEncode(primaryKey, lockVer)
+	iter := newIterator(mvcc.db, &util.Range{
+		Start: startKey,
+	})
+	defer iter.Release()
+
+	if iter.Valid() {
+		dec := lockDecoder{
+			expectKey: primaryKey,
+		}
+		ok, err := dec.Decode(iter)
+		if err != nil {
+			return 0, 0, errors.Trace(err)
+		}
+		// If current transaction's lock exists.
+		if ok && dec.lock.startTS == startTS {
+			lock := dec.lock
+			batch := &leveldb.Batch{}
+
+			// If the lock has already outdated, clean up it.
+			if uint64(oracle.ExtractPhysical(lock.startTS))+lock.ttl < uint64(oracle.ExtractPhysical(currentTS)) {
+				if err = rollbackLock(batch, lock, primaryKey, startTS); err != nil {
+					return 0, 0, errors.Trace(err)
+				}
+				if err = mvcc.db.Write(batch, nil); err != nil {
+					return 0, 0, errors.Trace(err)
+				}
+				return 0, 0, nil
+			}
+
+			// If this is a large transaction and the lock is active, push forward the minCommitTS.
+			// lock.minCommitTS == 0 may be a secondary lock, or not a large transaction.
+			if lock.minCommitTS > 0 {
+				lock.minCommitTS = currentTS
```

Comment:
According to the document, here it should be:
```
if lock.minCommitTs < currentTs + 1 {
    lock.minCommitTs = currentTs + 1;
    // write the lock...
}

Reply:
1. I haven't used the minCommitTS as the final commit ts yet, so it doesn't matter here.

2. The `currentTS` may not be a real current tso from the PD, it may be a local cached one in TiDB, it's inaccurate.

3. The real constraint here is `lock.minCommitTS >= startTS + 1`, so set minCommitTS to currentTS is ok as long as `currentTS >= startTS + 1`

I'll update here
@MyonKeminta

Reply:
We don't need to update the minCommitTS if minCommitTS is already greater than the startTS.

Reply:
OK, I'll add this condition as it saves a write operation some times.